src/hooks/useAuth.ts

```diff
@@ -97,9 +99,52 @@ export const useAuth = () => {
       return { error: firebaseError.message }
     } finally {
       setLoading(false);
+    }
+  }
+  return { firebaseRegister, firebaseLogIn, loading, error, isUserConnected, logOut };
+}
 
+export const useUserAuth = () => {
+  const [authUserInfo, setAuthUserInfo] = useState<UserType | User | null>(null);
+  const [authUserIsLoading, setAuthUserIsLoading] = useState<boolean>(true);
+
+  const getUserInfoFromFirestore = async (user: UserType | User) => {
+    if (auth.currentUser) {
+      // We fetch the connected user's info from the database using their ID
+      const docRef = doc(db, "users", auth.currentUser.uid);
+      const docSnap = await getDoc(docRef);
+      let userInfo = user
+  
+      if (docSnap.exists()) {        
+        // If the document exists (the user in the database), we fetch the desired data
+        userInfo = docSnap.data() as UserType
+      }
+      // Otherwise, we use the info from Firebase authentication
+      setAuthUserInfo(userInfo)
+      setAuthUserIsLoading(false)
     }
   }
 
-  return { firebaseRegister, firebaseLogIn, loading, error, isUserConnected, logOut };
+  const authStateChanged = async (currentUser: UserType | User | null) => {
+    if (!currentUser) {
+      setAuthUserInfo(null)
+      setAuthUserIsLoading(false)
+      return
+    }
+    setAuthUserIsLoading(true)
+    await getUserInfoFromFirestore(currentUser)
+  }
+
+  // Setting up a listener for changes in authentication state. 
+  useEffect(() => {
+    // When the authentication state changes, the callback function provided to 'onAuthStateChanged' (authStateChanged) is called with the current user's as a parameter. If no user is logged in, Firebase calls it with `null`.
+    const unsubscribe = onAuthStateChanged(auth, authStateChanged);
+    // Unsubscribe function returned by useEffect is called when the component is unmounted, which helps to avoid memory leaks.
+    return () => unsubscribe();
+  }, []);
```

Comment:
Attention, petit message d'alerte ici : 
"React Hook useEffect has a missing dependency: 'authStateChanged'. Either include it or remove the dependency array"